app/views/lookup_keys/_value.html.erb

```diff
@@ -10,9 +10,13 @@
   </td>
   <td>
     <div class="input-group">
-      <%= f.text_area :value, :rows => 1, :class => 'form-control no-stretch', :'data-property' => 'value',
-                      :disabled => f.object.use_puppet_default,
-                      :placeholder => _("Value") %>
+      <% if f.object.hidden_value? %>
+          <%= password_field_tag 'value', f.object.value, :class => 'form-control', :disabled => f.object.use_puppet_default %>
```

Comment:
Nitpick, this has four-space indentation between the if statement and password_field_tag, but should be two.